ui/appui-react/src/appui-react/content/FloatingViewportContent.tsx

```diff
@@ -39,11 +41,22 @@ export interface FloatingViewportContentProps {
  */
 // istanbul ignore next
 export function FloatingViewportContent(props: FloatingViewportContentProps) { // eslint-disable-line @typescript-eslint/naming-convention
-  const { contentId, initialViewState } = props;
+  const { contentId, initialViewState, viewportRef } = props;
   const [viewport, setViewport] = React.useState<ScreenViewport | undefined>();
   const contentControl = React.useRef<FloatingViewportContentControl | undefined>();
 
-  const onViewportRef = React.useCallback((v: ScreenViewport) => setViewport(v), []);
+  const viewState = React.useMemo(() => typeof initialViewState === "function" ? initialViewState() : initialViewState, [initialViewState]);
+  const onViewportRef = React.useCallback((v: ScreenViewport) => {
+    setViewport(v);
+    if (!viewportRef) {
+      return;
+    }
+    if (typeof viewportRef === "function") {
+      viewportRef(v);
+    } else {
+      viewportRef.current = v;
+    }
+  }, [viewportRef]);
```

Comment:
Would this work?
```suggestion
  const ref = React.useCallback((v: ScreenViewport) => {
    setViewport(v);
  }, []);
  const onViewportRef = useRefs(ref, ...(viewportRef ? [viewportRef] : []));
```

Reply:
We should probably allow `undefined` in `useRefs` to simplify cases like these...